### ISSUE TITLE
STM32L0538DISCOVERY

### ISSUE THREAD

**Issue by WestfW** [texane/stlink#283]
Any ideas why stlink fails to recognize by STM32L053C8DISCOVERY board (that's the one with the e-paper display), but works just fine on a Nucleo-L053R8 ?  (They have cpus from the same family; the discovery has a 48-pin version and the nucleo has a 64pin chip.  The Reference manual says all of the L0x3 chips should have the same chipId.)

here's the output from the discovery:
````
./st-util -v99
2014-11-25T08:10:04 DEBUG src/stlink-common.c: stlink current mode: debug (jtag or swd)
2014-11-25T08:10:04 DEBUG src/stlink-common.c: stlink current mode: debug (jtag or swd)
2014-11-25T08:10:04 DEBUG src/stlink-common.c: *** looking up stlink version
2014-11-25T08:10:04 DEBUG src/stlink-common.c: st vid         = 0x0483 (expect 0x0483)
2014-11-25T08:10:04 DEBUG src/stlink-common.c: stlink pid     = 0x374b
2014-11-25T08:10:04 DEBUG src/stlink-common.c: stlink version = 0x2
2014-11-25T08:10:04 DEBUG src/stlink-common.c: jtag version   = 0x15
2014-11-25T08:10:04 DEBUG src/stlink-common.c: swim version   = 0x5
2014-11-25T08:10:04 INFO src/stlink-common.c: Loading device parameters....
2014-11-25T08:10:04 DEBUG src/stlink-common.c: *** stlink_core_id ***
2014-11-25T08:10:04 DEBUG src/stlink-common.c: core_id = 0x00000000
2014-11-25T08:10:04 DEBUG src/stlink-common.c: *** stlink_read_debug32 e0042000 is 0xe0042000
2014-11-25T08:10:04 WARN src/stlink-common.c: unknown chip id! 0xe0042000
2014-11-25T08:10:04 DEBUG src/stlink-common.c: *** stlink_close ***
````
And, from the Nucleo:
````
./st-util -v99
2014-11-25T17:20:17 DEBUG src/stlink-common.c: stlink current mode: mass
2014-11-25T17:20:17 DEBUG src/stlink-common.c: stlink current mode: mass
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** stlink_enter_swd_mode ***
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** looking up stlink version
2014-11-25T17:20:17 DEBUG src/stlink-common.c: st vid         = 0x0483 (expect 0x0483)
2014-11-25T17:20:17 DEBUG src/stlink-common.c: stlink pid     = 0x374b
2014-11-25T17:20:17 DEBUG src/stlink-common.c: stlink version = 0x2
2014-11-25T17:20:17 DEBUG src/stlink-common.c: jtag version   = 0x16
2014-11-25T17:20:17 DEBUG src/stlink-common.c: swim version   = 0x5
2014-11-25T17:20:17 INFO src/stlink-common.c: Loading device parameters....
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** stlink_core_id ***
2014-11-25T17:20:17 DEBUG src/stlink-common.c: core_id = 0x0bc11477
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** stlink_read_debug32 0 is 0xe0042000
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** stlink_read_debug32 10086417 is 0x40015800
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** stlink_read_debug32 39f0040 is 0x1ff8007c
2014-11-25T17:20:17 INFO src/stlink-common.c: Device connected is: L0x3 device, id 0x10086417
2014-11-25T17:20:17 INFO src/stlink-common.c: SRAM size: 0x2000 bytes (8 KiB), Flash: 0x10000 bytes (64 KiB) in pages of 128 bytes
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** stlink_reset ***
2014-11-25T17:20:17 INFO gdbserver/gdb-server.c: Chip ID is 00000417, Core ID is  0bc11477.
2014-11-25T17:20:17 DEBUG src/stlink-common.c: *** reading target voltage
2014-11-25T17:20:17 DEBUG src/stlink-common.c: target voltage = 3251mV
2014-11-25T17:20:17 INFO gdbserver/gdb-server.c: Target voltage is 3251 mV.
2014-11-25T17:20:17 INFO gdbserver/gdb-server.c: Listening at *:4242...
````

Hmm.  I see that the JTAG version is different?

**Comment by xor-gate** [comment]
I'm digging up old issues, have you tried with latest programmer firmware from ST and latest stlink tools from this project?

**Comment by xor-gate** [comment]
This problem should be solved in `v1.2.0` or `master`. I'm closing due to communication inactivity. Feel free to open a new issue.
